### PR TITLE
[Graphbolt] Convert minibatch to dgl block instead of graph

### PR DESCRIPTION
**Pull request by peizhou001**
## Description
 Convert minibatch to dgl block instead of graph.

## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: python/dgl/graphbolt/minibatch.py [hunk 1]
```diff
@@ -126,89 +126,101 @@ class MiniBatch:
...
-        the graphs.
+        the blocks.
         """
         if not self.sampled_subgraphs:
             return None
 
+        reverse_row_node_ids = self.sampled_subgraphs[0].reverse_row_node_ids
+        assert (
+            reverse_row_node_ids is not None,
+            "Missing `reverse_row_node_ids` in sampled sub graph.",
```
Comment (Rhett-Ying) [inline python/dgl/graphbolt/minibatch.py:140]: `sub-graph` or `subgraph`.

File: python/dgl/graphbolt/minibatch.py [hunk 2]
```diff
@@ -126,89 +126,101 @@ class MiniBatch:
...
+        )
+        reverse_column_node_ids = self.sampled_subgraphs[
+            0
```
Comment (Rhett-Ying) [inline python/dgl/graphbolt/minibatch.py:143]: not `-1`? always fetch from first subgraph?

File: tests/python/pytorch/graphbolt/impl/test_to_dgl_blocks.py [hunk 3]
```diff
@@ -23,47 +22,46 @@ def test_to_dgl_graphs_hetero():
...
         reverse_edge_ids=reverse_edge_ids,
     )
-    g = gb.MiniBatch(
+    b = gb.MiniBatch(
         sampled_subgraphs=[subgraph],
         node_features=node_features,
         edge_features=[edge_features],
-    ).to_dgl_graphs()[0]
+    )
+    b = b.to_dgl_blocks()[0]
```
Comment (Rhett-Ying) [inline tests/python/pytorch/graphbolt/impl/test_to_dgl_blocks.py:30]: add testcases for multiple subgraphs?

File: python/dgl/graphbolt/minibatch.py (hunk 2 above)
Comment (peizhou001) [reply python/dgl/graphbolt/minibatch.py:143]: _1 is the `innermost`, 0 is `outermost`

File: tests/python/pytorch/graphbolt/impl/test_minibatch.py [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+    }
+    edge_features = {(relation, "x"): torch.randint(0, 10, (3,))}
+    subgraph = gb.SampledSubgraphImpl(
+        node_pairs=node_pairs,
+        reverse_column_node_ids=reverse_column_node_ids,
+        reverse_row_node_ids=reverse_row_node_ids,
+        reverse_edge_ids=reverse_edge_ids,
+    )
+    blocks = gb.MiniBatch(
+        sampled_subgraphs=[subgraph, subgraph],
```
Comment (frozenbugs) [inline tests/python/pytorch/graphbolt/impl/test_minibatch.py:26]: Let's spend slightly more effort here to make the data correct. The 2 subgraph should be different, and it is better to show that the dst of the first layer is the src of the second layer.

File: python/dgl/graphbolt/minibatch.py [hunk 5]
```diff
@@ -134,45 +134,45 @@ def to_dgl_blocks(self):
...
-                for subgraph in self.sampled_subgraphs
-            ]
+            else:
+                blocks.append(
+                    dgl.create_block(
+                        subgraph.node_pairs,
+                        num_src_nodes=reverse_row_node_ids.size(0),
+                        num_dst_nodes=reverse_column_node_ids.size(0),
+                    )
+                )
```
Comment (frozenbugs) [inline python/dgl/graphbolt/minibatch.py:175]: Can we do:

```
if is_heterogeneous:
  node_pairs = ...
  num_src_nodes = ...
  num_dst_nodes = ...
else:
  node_pairs = ...
  num_src_nodes = ...
  num_dst_nodes = ...
blocks.append(dgl.create_block(node_pairs, num_src_nodes, num_dst_nodes))
```

**Pull request merged**